cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java

```diff
@@ -328,6 +332,11 @@ public int getFirstTag() {
      */
     protected int _typeByte;
 
+    /**
+     * A pointer to know where to write text when we share an output buffer across methods
+     */
+    protected int _sharedOutBufferPtr;
```

Comment:
No, should not add this as state -- pointer should be passed along as needed (along with output buffer itself)

Reply:
Yeah, I didn't like that either, the problem is that the method that make use of it need to return this value, the pointer to the current buffer (that can change inside the method is it's replace after being full), along side with whatever value needs to respond back to its caller (e.g., boolean for success).
Initially I created a small static class for this, e.g.,

```
static class OutBufState {
   char[] _outBuf;
   int _outBufPtr;
}
```

If this seems better I can go back to that option. I like it better but wasn't sure about introducing a new class inside this one. Let me know.

Reply:
Ah. Yes, that makes sense, was guessing there has to be a reason.

And I think adding class is bit more intrusive. Let me think about this a bit and see.

Reply:
@sugmanue Ok, one other thought before giving up on this -- `TextBuffer` already has `_currentSize` to go with `_currentSegment`:

```
    public int getCurrentSegmentSize() { return _currentSize; }
    public void setCurrentLength(int len) { _currentSize = len; }
```

so perhaps that could be used instead, to sync output pointer?

Reply:
Sure, I will take a look. I tried with `_inputStart` which seemed a good candidate but it has different semantics. I will check and let you know. Thanks

Reply:
@cowtowncoder that did work perfectly. Thanks for the tip, I just pushed another commit with the changes, sorry for the delay, let me know what you think.

Reply:
@sugmanue Excellent! I'll have a look and hopefully can now just merge this, then up to `master` (for 3.0)